packages/pods-core/src/builtin/GDocPod.ts

```diff
@@ -225,40 +236,62 @@ export class GDocImportPod extends ImportPod<GDocImportPodConfig> {
     return note;
   }
 
-  createNote= async(
+  createNote= async( opts : {
     note: NoteProps,
     engine: DEngineClient,
     wsRoot: string,
-    vault: DVault
+    vault: DVault,
+    confirmOverwrite?: boolean,
+    onPrompt?: (arg0?: String) => Promise<{title : string} | undefined>
+  }
   ) => {
+     const { note, engine, wsRoot, vault, confirmOverwrite, onPrompt } = opts;
       const existingNote = NoteUtils.getNoteByFnameV5({
       fname: note.fname,
       notes: engine.notes,
       vault,
       wsRoot,
     });
     if (!_.isUndefined(existingNote)) {
-      if(existingNote.custom.revisionId && existingNote.custom.revisionId !== note.custom.revisionId) {
-        existingNote.custom.revisionId = note.custom.revisionId;
-        existingNote.body = note.body;
-        await engine.writeNote(existingNote, { newNode: true });
-        }   
-       }
+        if(existingNote.custom.revisionId && existingNote.custom.revisionId !== note.custom.revisionId) {
+          existingNote.custom.revisionId = note.custom.revisionId;
+          existingNote.body = note.body;
+
+          if(confirmOverwrite && onPrompt){
+          const resp = await onPrompt("userPrompt");
+
+            if(resp?.title.toLowerCase() === "yes"){
+              await engine.writeNote(existingNote, { newNode: true });
+              return existingNote;
+
+            }  
+          }
+          else{
+            await engine.writeNote(existingNote, { newNode: true });
+            return existingNote;
+          }
+        }else if(onPrompt){
```

Comment:
Do we expect onPrompt to be set without confirmOverwrite set? Are they meant to be set together?

Reply:
We expect  onPrompt to be set without confirmOverwrite set. Even if the confirmOverwrite is set to false, if the user imports the gdoc again and the revision id of gdoc and imported note is same, the user will be prompted as `Note is already in sync with the google doc`